feat: add documentation about assume role support for rds iam authentication (#7651)
Kong/docs.konghq.com: app/_src/gateway/kong-enterprise/aws-iam-auth-to-rds-database.md

   
   {:.warning}
   > **Warning:** You **can't** change the value of the environment variables you used to provide the AWS credential after booting {{site.base_gateway}}. Any changes are ignored.
{% if_version gte:3.8.x %}

>    - {{site.base_gateway}} also supports role assuming which allows you to use a different IAM role to authenticate to the database. If you want to use role assuming, make sure the original IAM role Kong uses has the correct permission to do role assuming to the target IAM role, and the target IAM role has the correct permission to connect to the database using IAM authentication.

Reviewer: ```suggestion
   - If you want to assume a role, make sure the original IAM role that Kong uses has the correct permission to assume the role of the target IAM role, and the target IAM role has the correct permission to connect to the database using IAM authentication.
```
You don't need the first sentence, this entry even being here implies that Kong Gateway supports it.

Rephrased to use more common language around AssumeRole. "Role assuming" sounds like it could be right, but AWS doesn't use that phrasing in their docs.

Author: > Rephrased to use more common language around AssumeRole. "Role assuming" sounds like it could be right, but AWS doesn't use that phrasing in their docs.
Good point! I'll be careful with that next time

KONG_PG_RO_IAM_AUTH=on
```

{% if_version gte:3.8.x %}

> If you want to use role assuming, set the following environment variables as well:

Reviewer: ```suggestion
If you want to [assume a role](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html), also set the following environment variables:
```

```bash

> # For read-write connection

Reviewer: ```suggestion
# For read-write connections
```

KONG_PG_IAM_AUTH_ASSUME_ROLE_ARN=<role_arn>
KONG_PG_IAM_AUTH_ROLE_SESSION_NAME=<role_session_name>

> # For read-only connection, if you need a different role for read-only connection

Reviewer: ```suggestion
# For read-only connections, if you need a different role than for read-write
```

pg_ro_iam_auth=on
```

{% if_version gte:3.8.x %}

> If you want to use role assuming, set the following environment variables as well:

Reviewer: ```suggestion
If you want to [assume a role](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html), also set the following configuration parameters:
```

```bash

> # For read-write connection

Reviewer: ```suggestion
# For read-write connections
```

pg_iam_auth_assume_role_arn=<role_arn>
pg_iam_auth_role_session_name=<role_session_name>

> # For read-only connection, if you need a different role for read-only connection

Reviewer: ```suggestion
# For read-only connections, if you need a different role than for read-write
```
